Sources/Schema.swift

```diff
@@ -100,7 +101,27 @@ extension SchemaBuilder {
         }
 
         let name = components[2]
+
+        if let cachedSchema = ReferenceResolver.cachedSchemaReferences[name] {
+            return Structure(name: name, structure: cachedSchema)
+        }
+        if let resolving = ReferenceResolver.resolvingReferences[name], resolving {
+            return Structure(name: name, structure: .unresolvedReference(pointer.path))
```

Comment:
You're right that it's not a valid spec, this value is just a placeholder during the first pass. It lets the resolver return _something_ if the resolving reference is currently resolving (thus stopping infinite recursion). I agree it shouldn't be part of the public API which is why I don't think this is the best implementation of this fix, but I can't see any other easy solution